src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8083,7 +8083,27 @@ BoundExpression bindInlineArrayElementAccess(ExpressionSyntax node, BoundExpress
                     }
                 }
 
-                // PROTOTYPE(InlineArrays): Check bounds
+                // Check bounds
+                if (convertedIndex.ConstantValueOpt is { SpecialType: SpecialType.System_Int32, Int32Value: int constIndex })
```

Comment:
Does this cover integer constants of other types, e.g., `Int16`?

Reply:
> Does this cover integer constants of other types, e.g., Int16?

We do not need to cover scenarios like that here. We can end up here only with three types: int, System.Index and System.Range.  See ``tryImplicitConversionToInlineArrayIndex````, for example.